packages/builders/src/components/fileUpload/FileUpload.ts

```diff
@@ -0,0 +1,109 @@
+import type { APIFileUploadComponent } from 'discord-api-types/v10';
+import { ComponentType } from 'discord-api-types/v10';
+import { validate } from '../../util/validation.js';
+import { ComponentBuilder } from '../Component.js';
+import { fileUploadPredicate } from './Assertions.js';
+
+/**
+ * A builder that creates API-compatible JSON data for file uploads.
+ */
+export class FileUploadBuilder extends ComponentBuilder<APIFileUploadComponent> {
+	/**
+	 * @internal
+	 */
+	protected readonly data: Partial<APIFileUploadComponent>;
+
+	/**
+	 * Creates a new file upload.
+	 *
+	 * @param data - The API data to create this file upload with
+	 * @example
+	 * Creating a file upload from an API data object:
+	 * ```ts
+	 * const fileUpload = new FileUploadBuilder({
+	 * 	custom_id: "file_upload",
+	 *  min_values: 2,
+	 *  max_values: 5,
+	 * });
+	 * ```
+	 * @example
+	 * Creating a file upload using setters and API data:
+	 * ```ts
+	 * const fileUpload = new FileUploadBuilder({
+	 * 	custom_id: "file_upload",
+	 *  min_values: 2,
+	 *  max_values: 5,
+	 * }).setRequired();
+	 * ```
+	 */
+	public constructor(data: Partial<APIFileUploadComponent> = {}) {
+		super();
+		this.data = { ...structuredClone(data), type: ComponentType.FileUpload };
+	}
+
+	/**
+	 * Sets the custom id for this file upload.
+	 *
+	 * @param customId - The custom id to use
+	 */
+	public setCustomId(customId: string) {
+		this.data.custom_id = customId;
+		return this;
+	}
+
+	/**
+	 * Sets the minimum number of file uploads required.
+	 *
+	 * @param minValues - The minimum values that must be uploaded
+	 */
+	public setMinValues(minValues: number) {
+		this.data.min_values = minValues;
+		return this;
+	}
+
+	/**
+	 * Clears the minimum values.
+	 */
+	public clearMinValues() {
+		this.data.min_values = undefined;
+		return this;
+	}
+
```

Comment:
```suggestion
```

I don't think this should be here. If you want to revert your change to minValues it's simply `setMinValues(1)` and should be explicit.

Reply:
See my other comment:
> We also don't do this on SelectMenuBuilders.

so no, we don't. We have it for things where there's an actual difference between providing a specific value or no value at all (`null`/`undefined`).

Reply:
Text inputs can be used in modals and they may clear their minimum and maximum length. Should that be removed then?